core/src/main/scala/akka/persistence/jdbc/state/scaladsl/JdbcDurableStateStoreQuery.scala

```diff
@@ -30,7 +33,29 @@ class JdbcDurableStateStoreQuery[A](
     case _             => ??? // should not reach here
   }
 
-  def changes(tag: String, offset: Offset): Source[DurableStateChange[A], NotUsed] = ???
+  def changes(tag: String, offset: Offset): Source[DurableStateChange[A], NotUsed] = {
+
+    Source
+      .unfoldAsync[(Offset, FlowControl), Seq[DurableStateChange[A]]]((offset, Continue)) { case (from, control) =>
+        def retrieveNextBatch() = {
+          /* get all records from the specified offset */
+          currentChanges(tag, from).runWith(Sink.seq).map { chgs =>
+            val nextStartOffset: Long = chgs.map { chg =>
+              chg.offset match {
+                case NoOffset      => 0
+                case APSequence(l) => l
+                case _             => 0 // should not reach here
+              }
+            }.max
+            /* if retrieved set is empty, then wait else continue */
+            val nextControl = if (chgs.isEmpty) ContinueDelayed else Continue
+            Some(((Offset.sequence(nextStartOffset), nextControl), chgs))
+          }
+        }
+        retrieveNextBatch()
+      }
+      .mapConcat(identity)
+  }
```

Comment:
I don't see that `control` is used anywhere, how does that `ContinueDelayed` delaying work, kind of looks like it will now just hit the `currentChanges` as fast as it can if there is downstream demand but no new results?

Reply:
Hi @johanandren - Possibly you are looking at an outdated commit .. I see this in the branch .. https://github.com/akka/akka-persistence-jdbc/blob/wip-durable-state/core/src/main/scala/akka/persistence/jdbc/state/scaladsl/JdbcDurableStateStore.scala#L114-L119

Reply:
Ah, that explains it, sorry for the noise, got tricked by github!